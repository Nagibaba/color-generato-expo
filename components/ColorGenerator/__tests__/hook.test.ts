import { act } from '@testing-library/react-native';
import { renderHook } from '@testing-library/react-hooks';

import useContainer from '../hook';

describe('ColorGenerator hook', () => {
  let result: any = null;
  beforeEach(() => {
    ({ result } = renderHook(() => useContainer()));

    jest.clearAllMocks();
  });

  it(`matches snapshot`, () => {
    expect(result.current).toMatchSnapshot();
  });

  it(`initial color is different from any hex color`, () => {
    const color = result.current.color;
    expect(color).toBe('white');
  });

  it(`running generateColor changes the color`, () => {
    const color = result.current.color;

    act(result.current.generateColor);

    expect(result.current.color).not.toBe(color);
  });
});
