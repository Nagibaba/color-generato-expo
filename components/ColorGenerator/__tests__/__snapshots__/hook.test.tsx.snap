// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorGenerator hook matches snapshot 1`] = `
{
  "color": "white",
  "generateColor": [Function],
}
`;
